types/express-serve-static-core/index.d.ts

```diff
@@ -61,7 +61,7 @@ export interface RequestHandler<
         req: Request<P, ResBody, ReqBody, ReqQuery, LocalsObj>,
         res: Response<ResBody, LocalsObj>,
         next: NextFunction,
-    ): void | Promise<void>;
+    ): unknown | Promise<unknown>;
```

Comment:
I am not sure why the tears aren't failing with this. This is equivalent to writing "unknown" because the union is reduced. I don't think this is a good type to use.

Reply:
see my previous comment

>  Note that we can't just use unknown in case people (like me) using eslint @typescript-eslint/no-misused-promises rule (error: Promise returned in function argument where a void return was expected)

From a strictly TypeScript perspective, you're correct; we could even have kept `void` as in v4 since `void` doesn’t limit the return type, allowing `Promise` to be returned with `void` or `unknown` alone. However, this approach can be cumbersome for those using `@typescript-eslint/no-misused-promises`, which requires the return type to explicitly reference `Promise`.

Reply:
I understand, but that sure seems like a linter bug. `unknown | number` is `unknown`. The types are eaten. 
![image](https://github.com/user-attachments/assets/39de9031-813e-4b3c-a44d-cd675b875e20)

I'm not sure how they can detect this unless they are somehow reading out the AST for the originally declared type, which will only ever work for hand-written code. If you wrote this code in TS today, it's absolutely free to not reprint this type as a union.

Reply:
See also from ts-eslint, https://typescript-eslint.io/rules/no-redundant-type-constituents/

Reply:
> I'm not sure how they can detect this unless they are somehow reading out the AST for the originally declared type, which will only ever work for hand-written code. If you wrote this code in TS today, it's absolutely free to not reprint this type as a union.

you're right, i double checked my test, and i should have done something wrong previously. Returning a promise on a function declared has `unknown` is not reported from `@typescript-eslint/no-misused-promises`.
I'm changing the PR